File: src/main/java/me/chriss99/ColoredVAO.java
package me.chriss99;

import me.chriss99.glabstractions.VAOImpl;

public class ColoredVAO extends VAOImpl {
    public ColoredVAO(double[] triangle, double[] color, int[] index) {
        super(index, 3, triangle, color);
    }

    public void updatePositions(double[] positions) {
        updateVertices(0, positions);
    }

    public void updateColors(double[] colors) {
        updateVertices(1, colors);
    }

    public void updateIndex(int[] index) {
        updateIndices(index);
    }
}

File: src/main/kotlin/render/ColoredVAO.kt
package me.chriss99.render

import me.chriss99.glabstractions.VAOImpl

open class ColoredVAO(triangle: DoubleArray, color: DoubleArray, index: IntArray) :
    VAOImpl(index, 3, triangle, color) {

    fun updatePositions(positions: DoubleArray) = updateVertices(0, positions)
    fun updateColors(colors: DoubleArray) = updateVertices(1, colors)
    fun updateIndex(index: IntArray) = updateIndices(index)
}